core/src/banking_stage.rs

```diff
@@ -443,15 +447,15 @@ impl BankingStage {
     fn record_transactions(
         bank_slot: Slot,
         txs: &[Transaction],
-        results: &[transaction::Result<()>],
+        results: &[TransactionProcessResult],
```

Comment:
The `TransactionProcessResult` plumbing in this PR is a great example of something that could be factored out and landed immediately.     It's cluttering up the meat of this PR and it barely even needs review

Reply:
Sure! I can do a bit of history cleanup

Reply:
For next time, please just bust this out immediately into a completely different PR and get it out of the way.   I really want to review this PR but at ~2k lines of new code I just can't do it.   20 smaller PRs that gradually introduce the feature, no problem!

Reply:
I second. Considering how long this PR has been open, even 40 PRs wouldn't have been a surprise.

Reply:
Thirded!  Admittedly not my best showing and full apologies for that!  I'm not sure I could've got it up to 40 independent parts, but even the four commits that it is now would've been much better

Reply:
Forthed!